### PR TITLE
Support for multiple rsids in VCF ID column

### PR DESCRIPTION
**Pull request by mcgml**
Great project. Just a tweak for your consideration to allow for multiple rsids in ID column which is part of the spec

### CODE REVIEW

**Comment by standage** [comment]
Hi @mcgml, sorry for my delayed response to this.

This PR is great! The only issue I encountered was testing this on VCFs where an rsID is repeated in multiple records: see the attached file, which is a subset of chr9 from the 1000 Genomes Project Phase 3 data.

```
sqlite3.IntegrityError: UNIQUE constraint failed: rsid_to_coord.rsid
```

One possible way to address this: add `found = set()` to the beginning of the `parse_vcf()` function, and then check to see whether this rsid has been found before (see my suggested edits).
[chr9_99202590-99202690.vcf.txt](https://github.com/bioforensics/rsidx/files/4313957/chr9_99202590-99202690.vcf.txt)

File: rsidx/index.py [hunk 1]
```diff
@@ -17,10 +17,11 @@ def parse_vcf(vcfstream, updateint=1e6):
...
             continue
-        chromstr, posstr, rsid, *values = line.split('\t')
-        if not rsid.startswith('rs'):
-            continue
-        yield int(rsid[2:]), chromstr, int(posstr)
+        chromstr, posstr, rsids, *values = line.split('\t')
+        for rsid in rsids.split(";"):
+            if not rsid.startswith('rs'):
+                continue
+            yield int(rsid[2:]), chromstr, int(posstr)
```
Comment (standage) [inline rsidx/index.py]: Regarding duplicated rsid issue: try adding `found = set()` to the beginning of the function and then adding this.

```python
        chromstr, posstr, rsids, *values = line.split('\t')
        for rsid in rsids.split(';'):
            if not rsid.startswith('rs'):
                continue
            rsid = int(rsid[2:])
            if rsid in found:
                continue
            yield rsid, chromstr, int(posstr)
            found.add(rsid)
```

**Comment by mcgml** [comment]
Hey thanks

Regarding duplicate RSIDs, I've encountered this before too but officially they are not permitted:

```
3. ID - identifier: Semi-colon separated list of unique identifiers where available. If this is a dbSNP variant it is
encouraged to use the rs number(s). **No identifier should be present in more than one data record**. If there is no
identifier available, then the missing value should be used. (String, no white-space or semi-colons permitted)
```

So one option is to fail, but if you want the flexibility then your idea to use a ```set()``` sounds great

Thanks
Matt

**Comment by mcgml** [comment]
@standage how about updating the SQL statement to:

```
INSERT OR IGNORE INTO rsid_to_coord VALUES
```

which is more memory efficient than storing all rsids?

**Comment by standage** [comment]
Looks great! I've added some tests, but since I can't push to the master branch of your fork, I'll handle it in a separate PR. Thanks!

**Pull request merged**